code/game/machinery/status_display.dm

```diff
@@ -267,6 +267,10 @@
 	name = "shuttle display"
 	var/shuttle_id
 
+/obj/machinery/status_display/shuttle/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_CONNECT_TO_SHUTTLE, .proc/connect_to_shuttle)

```

Comment:
I'm really unsure about allowing things to listen for signals on themselves, why isnt this just a regular `/atom` proc?

Reply:
To avoid proc call overhead for every single atom in every shuttle

Reply:
SEND_SIGNAL doesn't call a proc if there's nothing listening but it does both check if a list exists and check if a key exists. This is also additional memory cost from the callback object.

Also I'd maybe want to move the baseturf changes to turfs into this functionality instead of it being shoved into template loading.